app/uk/gov/hmrc/helptosavefrontend/services/HelpToSaveService.scala

```diff
@@ -125,6 +128,21 @@ class HelpToSaveServiceImpl @Inject() (helpToSaveConnector: HelpToSaveConnector)
     }
     )
 
+  override def validateBankDetails(barsRequest: BarsRequest)(implicit hc: HeaderCarrier, ex: ExecutionContext): Result[Boolean] =
+    EitherT(helpToSaveConnector.validateBankDetails(barsRequest).map[Either[String, Boolean]] { response ⇒
+      response.status match {
+        case Status.OK ⇒
+          Try((response.json \ "isValid").as[Boolean]) match {
+            case Success(isvalid) ⇒ Right(isvalid)
+            case Failure(error)   ⇒ Left(s"couldn't parse bars response from BE, error=${error.getMessage}. Body was ${maskNino(response.body)}")
+          }
+
+        case other ⇒
+          Left(s"Received unexpected status $other from BARS validation. Body was ${maskNino(response.body)}")
```

Comment:
here and elsewhere in the FE I'd refrain from mentioning BARs and instead try to use something more generic like BankDetailsValidation or something - the BE should be the only thing concerned with how the bank details are validated, the FE doesn't need to know about BARs. That way, if we ever swap out BARs for something else, the FE code and logs would still make sense